orttraining/orttraining/core/graph/gradient_builder_base.h

```diff
@@ -57,6 +58,15 @@ class GradientBuilderBase {
   // i-th input of forward op
   ArgDef I(const size_t i) const {
     ORT_ENFORCE(i < node_->InputDefs().size());
+
+    const std::string& name = node_->InputDefs()[i]->Name();
+    NodeArg* recomputed_nodearg = graph_->GetNodeArg(name + "_recompute");
```

Comment:
>graph_ [](start = 34, length = 6)

If we pass-in explicit node_arg mapping, graph is no longer needed

Reply:
idea aborted. 

---
In reply to: [448740047](https://github.com/microsoft/onnxruntime/pull/4402#discussion_r448740047) [](ancestors = 448740047)